scripts/scaletest.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+
+if [[ $# -lt 2 ]]; then
+	echo "Usage: $0 <name> <scenario> <num_workspaces>"
+fi
+
+[[ -n ${VERBOSE:-} ]] && set -x
+set -euo pipefail
+
+SCALETEST_NAME="$1"
+SCALETEST_SCENARIO="$2"
+SCALETEST_NUM_WORKSPACES="$3"
+SCALETEST_PROJECT="${SCALETEST_PROJECT:-}"
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+SCALETEST_SCENARIO_VARS="${PROJECT_ROOT}/scaletest/terraform/scenario-${SCALETEST_SCENARIO}.tfvars"
+SCALETEST_SECRETS="${PROJECT_ROOT}/scaletest/terraform/secrets.tfvars"
+SCALETEST_SECRETS_TEMPLATE="${PROJECT_ROOT}/scaletest/terraform/secrets.tfvars.tpl"
+SCALETEST_SKIP_CLEANUP="${SCALETEST_SKIP_CLEANUP:-}"
+
+if [[ "${SCALETEST_SKIP_CLEANUP}" == "true" ]]; then
+	echo "WARNING: you told me to not clean up after myself, so this is now your job!"
+fi
+
+if [[ -z "${SCALETEST_PROJECT}" ]]; then
+	echo "Environment variable SCALETEST_PROJECT not set. Please set it and try again."
+	exit 1
+fi
+
+if [[ ! -f "${SCALETEST_SCENARIO_VARS}" ]]; then
+	echo "No definition for scenario ${SCALETEST_SCENARIO} exists. Please create it and try again"
+	exit 1
+fi
+
+echo "Writing scaletest secrets to file."
+SCALETEST_NAME="${SCALETEST_NAME}" envsubst <"${SCALETEST_SECRETS_TEMPLATE}" >"${SCALETEST_SECRETS}"
+
+pushd "${PROJECT_ROOT}/scaletest/terraform"
+
+echo "Initializing terraform."
+terraform init
+
+echo "Setting up infrastructure."
+terraform plan --var-file="${SCALETEST_SCENARIO_VARS}" --var-file="${SCALETEST_SECRETS}" -out=scaletest.tfplan
+terraform apply -auto-approve scaletest.tfplan
+
+SCALETEST_CODER_URL=$(<./.coderv2/url)
+attempt_counter=0
+max_attempts=6 # 60 seconds
+echo -n "Waiting for Coder deployment at ${SCALETEST_CODER_URL} to become ready"
+until curl --output /dev/null --silent --fail "${SCALETEST_CODER_URL}/healthz"; do
```

Comment:
I'm not a fan of `bash` for this sort of thing because it makes it basically impossible to be cross-platform due to shelling out to commands that we don't control the version of.

Reply:
It's not `curl`, _in particular_ that's the problem...